org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventConditions.java

```diff
@@ -0,0 +1,829 @@
+		return VerboseCondition.verboseCondition((be) -> be.getOrigin() == null, "origin is <null>",
+			be -> "was " + be);
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link BundleEvent}. Checking
+	 * if a type-mask <b>matches</b> the {@link BundleEvent} type of lifecyle
+	 * event.
+	 *
+	 * <pre>
+	 * List<BundleEvent> bundleEvents = null;
+	 *
+	 * static void example_type(int typeMask) {
+	 *
+	 * 	assertThat(bundleEvents)// created an {@link ListAssert}
+	 * 		.have(type(typeMask))
+	 * 		.filteredOn(type(typeMask))
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(type(typeMask));// used on {@link ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @param typeMask - the typeMask that would be checked against the bundle
+	 *            type of the {@link BundleEvent}
+	 * @return the Condition
+	 */
+	static Condition<BundleEvent> type(final int expectedEventTypeMask) {
+		return VerboseCondition.verboseCondition(
+			(BundleEvent be) -> Bitmap.typeMatchesMask(be.getType(), expectedEventTypeMask),
+			"type matches mask <" + BundleEventType.BITMAP.maskToString(expectedEventTypeMask) + ">", //
+			be -> " was <" + BundleEventType.BITMAP.toString(be.getType()) + ">");
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link BundleEvent}. Checking
+	 * if a given type-mask <b>matches</b> the type <i>and</i> the given
+	 * {@link Bundle} <b>equals</b> the Bundle of the of lifecyle event.
+	 *
+	 * <pre>
+	 * List<BundleEvent> bundleEvents = null;
+	 *
+	 * static void example_typeAndBundle(int typeMask, Bundle bundle) {
+	 *
+	 * 	assertThat(bundleEvents)// created an {@link ListAssert}
+	 * 		.have(typeAndBundle(typeMask, bundle))
+	 * 		.filteredOn(typeAndBundle(typeMask, bundle))
+	 * 		.first()// map to
+	 * 				// {@link
+	 * 				// ObjectAssert}
+	 * 		.has(typeAndBundle(typeMask, bundle));// used on {@link
+	 * 												// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @param typeMask - the typeMask that would be checked against the bundle
+	 *            type of the {@link BundleEvent}
+	 * @param bundle - the bundle that would be checked against the bundle of
+	 *            the {@link BundleEvent}
+	 * @return the Condition
+	 */
+	static Condition<BundleEvent> typeAndBundle(int mask, Bundle bundle) {
+		return Assertions.allOf(type(mask), bundleEquals(bundle));
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link BundleEvent}. Checking
+	 * if the {@link BundleEvent} type <b>matches BundleEvent.INSTALLED</b>.
+	 *
+	 * <pre>
+	 * List<BundleEvent> bundleEvents = null;
+	 *
+	 * static void example_typeInstalled()) {
+	 *
+	 * 	assertThat(bundleEvents)// created an {@link ListAssert}
+	 * 		.have(typeInstalled()))
+	 * 		.filteredOn(typeInstalled()))
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(typeInstalled());// used on {@link ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	static Condition<BundleEvent> typeInstalled() {
```

Comment:
I don't think we want to have methods for each event type.

Reply:
reduced to just one `type(int mask)` .